### PR TITLE
Add readme templates

### PR DESCRIPTION
**Pull request by MichaelSimons**
Related to #dotnet/dotnet-docker#1931

* Created a single readme template to generate all of the readmes from
* Updated the unit tests to ensure the readmes and templates are always in sync
* Made one small tweak to the README.md to use the same formatting for the _Related Repos_ section.

### CODE REVIEW

File: eng/readme-templates/Use.monitor.md [hunk 1]
```diff
@@ -0,0 +1 @@
+TBD
```
Comment (mthalman) [inline eng/readme-templates/Use.monitor.md:1]: Is this intended to be TBD as part of this PR?

Comment (MichaelSimons) [reply eng/readme-templates/Use.monitor.md:1]: Yes - that is what the readme currently contains - https://github.com/dotnet/dotnet-docker/blob/nightly/README.monitor.md#how-to-use-the-image

File: eng/readme-templates/Get-GeneratedReadmes.ps1 [hunk 2]
```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env pwsh
+#!/usr/bin/env pwsh
```
[the remark this answers is not in the document]
Comment (MichaelSimons) [reply eng/readme-templates/Get-GeneratedReadmes.ps1:2]: Removed.

File: eng/readme-templates/README.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+# How to Use the Image{{if IS_PRODUCT_FAMILY:s}}
+
+The [.NET Core Docker samples](https://github.com/dotnet/dotnet-docker/blob/master/samples/README.md) show various ways to use .NET Core and Docker together. See [Building Docker Images for .NET Core Applications](https://docs.microsoft.com/dotnet/core/docker/building-net-docker-images) to learn more.
+
+{{InsertTemplate(join(filter(["Use", SHORT_REPO, "md"], len), "."))}}
+# Related Repos
+
+.NET Core 2.1/3.1:
+
+{{if !IS_PRODUCT_FAMILY || (IS_PRODUCT_FAMILY && VARIABLES["branch"] = "nightly")
```
[the remark this answers is not in the document]
Comment (MichaelSimons) [reply eng/readme-templates/README.md]: You are correct, I fixed all four occurrences of this pattern.

**Review by mthalman — APPROVED** [review]

**Pull request merged**